aws/resource_aws_cloudfront_distribution.go

```diff
@@ -478,6 +478,56 @@ func resourceAwsCloudFrontDistribution() *schema.Resource {
 					},
 				},
 			},
+			"origin_group": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      originGroupHash,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"origin_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.NoZeroValues,
+						},
+						"failover_criteria": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Set:      failoverCriteriaHash,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"status_codes": {
+										Type:     schema.TypeList,
+										Required: true,
+										Elem:     &schema.Schema{Type: schema.TypeInt},
+									},
+								},
+							},
+						},
+						"members": {
```

Comment:
To simplify the schema for operators while maintaining some consistency with how these Quantity/Items objects are handled in other parts of the resource, I would recommend removing the additional layer in the `members` configuration blocks so operator's configurations become:

```hcl
  origin_group {
    origin_id = "groupS3"

    failover_criteria {
      status_codes = [403, 404, 500, 502]
    }

    member {
      origin_id = "primaryS3"
    }

    member {
      origin_id = "failoverS3"
    }
  }
```

Something like this should work for the schema:

```go
"member": {
  Type:     schema.TypeList,
  Required: true,
  MinItems: 2,
  Elem: &schema.Resource{
    Schema: map[string]*schema.Schema{
      "origin_id": {
        Type:     schema.TypeString,
        Required: true,
      },
    },
  },
},
```

I think we can leave off the `ordered_` prefix in the attribute name if the ordering behavior is documented well (the `ordered_` naming prefix was an upgrade path for `cache_behavior` which was unordered previously).